Improve ElasticsearchTaskHandler

millin opened the pull request:
Improve ElasticsearchTaskHandler:
 - use builtin logging.makeLogRecord instead of strange _ESJsonLogFmt
 - do not re-sort already sorted logs
 - apply ISO 8601 datefmt, because ES defines it as a datetime, not a text
 - fixed several found bugs

potiuk commented on airflow/utils/log/json_formatter.py:
```diff
@@ -21,12 +21,18 @@
 import json
 import logging
 
+import pendulum
```
This is part of the airflow core.  This change changes elasticsearch provider so there are two problems: 

1) What happens if the new provider is installed with old airflow? Do we have some dependency here on those modifications? I guess yes because otherwise you would not not change it. Can you please make it standalone in the way that it will not require changes in airlfow core?

2) Secondly (related)- what's the impact of this change on the rest of Airlfow? How formatTime addition will impact any other place  that uses JSONFormatter? Did you try to assess the impact?

millin replied:
@potiuk I made a separate formatter for ES. So now it won't break compatibility

potiuk commented on airflow/www/static/js/ti_log.js:
```diff
@@ -99,7 +99,7 @@ function autoTailingLog(tryNumber, metadata = null, autoTailing = false) {
       // Detect urls and log timestamps
       const urlRegex = /http(s)?:\/\/[\w.-]+(\.?:[\w.-]+)*([/?#][\w\-._~:/?#[\]@!$&'()*+,;=.%]+)?/g;
-      const dateRegex = /\d{4}[./-]\d{2}[./-]\d{2} \d{2}:\d{2}:\d{2},\d{3}/g;
+      const dateRegex = /([^'])(\d{4}[./-]\d{2}[./-]\d{2})[T ](\d{2}:\d{2}:\d{2})(?:[.,](\d{3,6}))?(\+\d{2}:?\d{2})?([^'])/g;
```
1. Why do we change parts of the web UI when the change concerns only elasticsearch? What happens here if other loggers are used ? Will tha timpact them as well? How?

millin replied:
This new regex allows to use both formats given elasticsearch and other loggers

millin committed:
Improve ElasticsearchTaskHandler:
- use builtin logging.makeLogRecord instead of strange _ESJsonLogFmt
- do not re-sort already sorted logs
- apply ISO 8601 datetime format
- fixed several found bugs

millin committed:
Fix docstrings & small fix

potiuk reviewed (approved):
LGTM but I would love someone from the committers to have elasticsearch experience to take a look.
